src/main/scala/viper/gobra/reporting/StatsCollector.scala

```diff
@@ -0,0 +1,345 @@
+          case _ =>
+        }
+      case GobraEntityFailureMessage(taskName, _, e, info, _, time, cached) if !info.node.isInstanceOf[BuiltInMember] =>
+        Violation.violation(typeInfos.contains(taskName), "No type info available for stats reporter")
+        getMemberInformation(info.pnode, typeInfos(taskName)) match {
+          case GobraMemberInfo(pkgDir, pkg, memberName, args, isTrusted, isAbstract, isImported, false) =>
+            addResult(
+              pkgDir,
+              pkg,
+              memberName,
+              info.pnode.getClass.getSimpleName,
+              args,
+              ViperMemberEntry(e, taskName, time, e.getClass.getSimpleName, success = false, cached = cached, isImported, hasBody = viperMemberHasBody(e)),
+              isTrusted,
+              isAbstract)
+          case _ =>
+        }
+      case _ =>
+    }
+    // Pass message to next reporter
+    reporter.report(msg)
+  }
+
+  def getNumberOfCachedViperMembers: Int = memberMap.values.flatMap(_.viperMembers.values).count(_.cached)
+
+  def getNumberOfCacheableViperMembers: Int = memberMap.values.flatMap(_.viperMembers.values).count(member => member.hasBody && member.nodeType == "Method")
+
+  def writeJsonReportToFile(file: File): Unit = {
+    if((file.exists() && file.canWrite) || file.getParentFile.canWrite) {
+      FileUtils.writeStringToFile(file, getJsonReport(false), UTF_8)
+    }
+  }
+
+  def getJsonReport(shorten: Boolean): String = {
+    val json = "[\n" +
+      memberMap.map({ case (key, gobraMember) => s"""  {
+    "id": "$key",
+    "pkgDir": "${gobraMember.pkgDir}",
+    "pkg": "${gobraMember.pkg}",
+    "name": "${gobraMember.pkg}.${gobraMember.memberName}",
+    "args": "${gobraMember.args}",
+    "nodeType": "${gobraMember.nodeType}",
+    "trusted": ${gobraMember.isTrusted},
+    "abstract": ${gobraMember.isAbstract},
+    "viperMembers": [ """ + "\n" +
+          gobraMember.viperMembers.values.map(viperMember => s"""      {
+        "name": "${viperMemberName(viperMember.member)}",
+        "taskName": "${viperMember.taskName}",
+        "nodeType": "${viperMember.nodeType}",
+        "time": ${viperMember.time},
+        "success": ${viperMember.success},
+        "fromImport": ${viperMember.fromImport},
+        "hasBody": ${viperMember.hasBody},
+        "cached": ${viperMember.cached}
+      }""").mkString(", \n") + "\n    ],\n" + s"""    "dependencies": [""" + "\n" +
+        gobraMember.viperMembers.values
+          .flatMap(viperMember => getDependencies(viperMember.member)
+            .flatMap(dep => findGobraMemberByViperMemberName(viperMember.taskName, dep, gobraMember.pkgDir))
+          )
+          // Filter out dependencies on one self, since this information isn't very useful
+          .filter(_ != gobraMember)
+          .map(gobraMemberEntry => "        \"" + gobraMemberKey(gobraMemberEntry.pkgDir, gobraMemberEntry.pkg, gobraMemberEntry.memberName, gobraMemberEntry.args) + "\"")
```

Comment:
we can use a number of tabs to keep track of the indentation instead of having string literals with fixed spaces

Reply:
I changed the solution to work with prefixes for indentation. I don't know if this is what you had in mind or if you explicitly prefer tabs over spaces.

Reply:
The reason why I meant tabs is that you can easily tell the indentation of something by looking at the string literal, whereas when you have string literals with an arbitrary number of spaces you literally need to count them

Reply:
I moved the indentation to  a variable, this way it is much clearer how much it is indented and it is configurable at a single point, how it should be indented